src/mixpanel-core.js

```diff
@@ -1106,6 +1108,362 @@ MixpanelLib.prototype.track = addOptOutCheckMixpanelLib(function(event_name, pro
     return ret;
 });
 
+/**
+ * Initializes the heartbeat tracking system for the instance
+ * @private
+ */
+MixpanelLib.prototype._init_heartbeat = function() {
+    var self = this;
+
+    // Internal heartbeat state storage
+    this._heartbeat_timers = new Map();
+    this._heartbeat_unload_setup = false;
+
+    // Setup page unload handlers once
+    this._setup_heartbeat_unload_handlers();
+
+    /**
+     * Client-side aggregation for streaming analytics events like video watch time,
+     * podcast listen time, or other continuous interactions. Designed to be called
+     * in loops without exploding row counts.
+     *
+     * Heartbeat works by aggregating properties client-side until the event is flushed.
+     * Properties are merged intelligently:
+     * - Numbers are added together
+     * - Strings take the latest value
+     * - Objects are merged (latest overwrites)
+     * - Arrays have elements appended
+     *
+     * Events auto-flush after 30 seconds (configurable) or on page unload.
```

Comment:
Is that every 30 seconds or after 30 seconds of inactivity?

Maybe I've missed it/misunderstood, but I'm surprised this method doesn't have an option for the SDK to repeat heartbeats on a regular schedule, so you don't have to keep calling `heartbeat` over and over yourself e.g.
```js
const videoHeartbeat = mixpanel.heartbeat(
  'podcast_listen', 'episode_123',
  {platform: 'mobile'},
  {interval_seconds: 5},
);

// then when video stops:
videoHeartbeat.stop();
```
(I haven't thought through that signature too much ^^^ but something like that I imagine would let you just call start/stop methods and not worry about managing your own timers)